src/posix/platform/backbone.cpp

```diff
@@ -55,8 +57,22 @@ void platformBackboneInit(otInstance *aInstance, const char *aInterfaceName)
 
     otLogInfoPlat("Backbone interface is configured to %s (%d)", gBackboneNetifName, gBackboneNetifIndex);
 
-exit:
-    return;
+    sMulticastRoutingManager.Init(aInstance);
+}
+
+void platformBackboneUpdateFdSet(fd_set &aReadFdSet, int &aMaxFd)
+{
+    sMulticastRoutingManager.UpdateFdSet(aReadFdSet, aMaxFd);
```

Comment:
As discussed, It's possible to simplify the design according to a good kernel version, and I think we should update the current implementation which is designed according to work with a buggy kernel.

Reply:
@jwhui @librasungirl @bukepo 

According to our investigation, latest 5.x Linux kernel supports (*, G) multicasting, meantime 4.x Linux and macOS(BSD) does not support (*, G) multicasting, but only support (S, G) multicasting. 

What MLR feature needs for the PBBR is (*, G) multicast routing, thus we need to implement (*, G) multicast routing using (S, G) multicast routing in the POSIX platform code.

Even though it's possible to use a simpler implementation using (*, G) multicast routing provided by latest Linux kernel, I think **the current implementation is valuable because it's more portable**. It works on a wide range of Linux kernel versions and can be ported to support macOS.

Reply:
> I think we should update the current implementation which is designed according to work with a buggy kernel.

I am not sure if it's fair to say that the current implementation is designed to work with a buggy kernel.

The current implementation uses a set of APIs that are provided by many POSIX platforms, including Linux, macOS, BSD. 
It does not use [MRT6_ADD_MFC_PROXY](
https://github.com/torvalds/linux/blob/3d5e28bff7ad55aea081c1af516cc1c94a5eca7d/include/uapi/linux/mroute6.h#L32) which is only provided by latest Linux to add a (\*, G) MFC entry, which is not well documented and not proven to work (I had a version using MRT6_ADD_MFC_PROXY but it didn't work as intended).

That being said, I am not against a future enhancement that utilizes MRT6_ADD_MFC_PROXY for Linux kernel versions that has working  (\*, G) MFC. But we should keep this implementation for portability. 

Thoughts?

Reply:
It is a good suggestion to do investigation on (*,G) which has some obvious advantages.
I may suggest do it later as future optimization without blocking this PR review.

Thoughts?
@bukepo